.github/actions/log-metric/action.yml

```diff
@@ -0,0 +1,20 @@
+name: 'Log Metric'
+description: 'Log data point to a metric with the provided value. If the metric is not there, it will create one.'
+inputs:
+  metric-name:
+    description: 'target metric name'
+    required: true
+  value:
+    description: 'the value to log to the metric'
+    required: true
+runs:
+  using: 'composite'
+  steps:
+    - name: Configure AWS credentials
+      uses: aws-actions/configure-aws-credentials@v1
+      env:
+        AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
```

Comment:
Will this work? According to https://stackoverflow.com/questions/70098241/using-secrets-in-composite-actions-github:

> you need to pass these secrets via inputs. there's a good reason for that: imagine if an action you're calling could use your secrets without you passing them to the action.

Reply:
Looks like two main ways of passing secrets in composite actions are:

(1) Pass them as inputs (e.g. [configure-aws-credentials](https://github.com/aws-actions/configure-aws-credentials)):
```yml
      - name: Configure AWS credentials
        uses: aws-actions/configure-aws-credentials@v1
        with:
          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
```


(2) Pass them as env and reuse it (e.g. changesets/action):

```yml
      - name: Publish to @latest
        uses: changesets/action@b98cec97583b917ff1dc6179dd4d230d3e439894
        with:
          publish: yarn publish:latest
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
```

Though (2) is harder to work with because we're just using a simple composite action. If (1) is safe to do so, then I'm fine with keeping it simple and have it passed as an input.

Reply:
Tested in my personal fork. Secrets are masked even after they are passed to `inputs`, so we can safely use (1) 🎉 

<img width="479" alt="image" src="https://user-images.githubusercontent.com/11983489/180127445-67402b14-ed15-4fcc-aee5-710d833d847d.png">